### DISCUSSION TITLE
Hover over the pasted file or folder.

### DISCUSSION THREAD

**Discussion by ze0987** [sxyazi/yazi#1706] [1. Q&A] [answered]
Is it possible to hover over the pasted entry (the first one, if there are many, according to the current sorting method)?

Here's the 'select -> paste' in nnn for reference:

https://github.com/user-attachments/assets/e38ae295-1c69-41ed-aba0-e89a61536f97

**Comment by sxyazi** [comment, marked as answer]
I think because nnn is synchronous — users can't do any other actions until the copy operation is completed, so it can handle things this way. 

However, Yazi's copy operations are asynchronous — they run in the background. This means that during copying, the user may have done other actions, like moving the cursor and getting ready to press "d" to delete a file. If the background copy task finishes at that moment and resets the cursor, the user might accidentally select the wrong file to delete, which poses a certain safety risk. 

Additionally, Yazi can have multiple task queues running concurrently in the background, and if they finish one after the other, the cursor would be reset multiple times, and this depends on which task finishes first, rather than the order in which the user initiated the copy operations, which is not a good user experience.

**Reply by ze0987** [reply]
Thanks for the explanation. These are valid points. What do you think about cancelling the hover when the beginning of the operation is still in the queue and the user presses any key in the meantime? If the paste starts immediately, the position of the first created entry could be hovered over.

By the way, I don't know if this is typical for GUI file managers, but I noticed that GNOME Files highlights pasted items after each completed queue. But I agree, this is not a safe behavior for a keyboard driven file manager.

**Comment by sxyazi** [comment]
This is now supported in https://github.com/sxyazi/yazi/pull/3846 🎉🎉
